### PR TITLE
Fix preview snippet bugs

### PR DESCRIPTION
**Pull request by arnaudbroes**
Issue #1408 & #2526.

## Proposed changes

Fixes two bugs for the preview snippet title.

(1) Double % signs in the title break the title (#2526)
(2) Double site title on Posts page (#1408)

The former was happening because there was some code that was deleting everything in between two percentage signs (including the signs); the latter was happening because the title for the Posts page is already handled by get_aioseop_title().

In the meanwhile, I've also cleaned up the phpDocs and refactored a little bit.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- Improves existing code

## Checklist

- [x] I've selected the appropriate branch (ie x.y rather than master).
- [x] I've created an appropriate title, descriptive of what the PR does.
- [x] Travis-ci runs with no errors.

## Testing instructions

For #2526:
Add two % signs to the title of a new Post with some text in between each % sign and make sure the preview snippet displays the title correctly.

For #1408:
Set a static Posts page under Settings > Reading, navigate to it and make sure the preview snippet doesn't include the site title twice.

### CODE REVIEW

**Comment by arnaudbroes** [comment]
@EkoJR can you confirm that lines 3417-3424 which I deleted weren't used by anything?

**Comment by wpsmort** [comment]
This is tested and is good.

**Comment by EkoJR** [comment]
@arnaudbroes It appears that code is being used, and I believe it is a default to fallback on when there is no title.

To reproduce.
1. From the admin menu, under a post type, click on 'Add New'
2. Go to AIOSEOP Title, and type anything. Then delete it.
3. The title should still show a title, but doesn't show any title.

PR2526-before

Changes to...
PR2526-after

File: aioseop_class.php [hunk 1]
```diff
@@ -1022,48 +1022,54 @@ function custom_output_option( $buf, $args ) {
...
 		$title        = $info['title'];
 		$description  = $info['description'];
 		$keywords     = $info['keywords'];
 		$url          = $info['url'];
 		$title_format = $info['title_format'];
 		$category     = $info['category'];
-		$w            = $info['w'];
-		$p            = $info['p'];
+		$wp_query     = $info['w'];
+		$post         = $info['p'];
```
Comment (EkoJR) [inline aioseop_class.php]: Thank you for changing the names on these. This is much better!

File: aioseop_class.php [hunk 2]
```diff
@@ -1137,31 +1143,39 @@ public function get_title_format( $args ) {
...
+		 * Runs after we have processed the title format for the snippet preview is.
+		 *
+		 * @since 3.0.0
 		 */
 		do_action( 'aioseop_after_format_title' );
 
 		return $title_format;
 	}
 
+	function get_preview_snippet_title_helper( $title_format ) {
```
Comment (EkoJR) [inline aioseop_class.php]: Missing phpDoc for the function.

File: aioseop_class.php (hunk 1 above)
Comment (arnaudbroes) [reply aioseop_class.php]: Thanks, I agree! I originally had no idea what I was looking at and had to browse through the code to find what these variables are.

**Comment by arnaudbroes** [comment]
@EkoJR incredible catch (since that slipped past Steve and me during testing)! 

I was able to reproduce the issue and fix it, although it wasn't caused by the removal of those lines. This was happening because the JavaScript on the Edit page affects the part of the preview snippet title that is within the span element. I had moved the entire preview snippet title inside the span element,. The static parts (basically all macros besides %post_title%/%page_title%) had to be moved out of the span element again so that they remain static and aren't modified on user input.

**Comment by arnaudbroes** [comment]
@wpsmort this needs to be retested. You'll now also want to test that the static parts of the title (all macros and input in the relevant title format field, e.g. %site_title%, with the exception of the page/post title) are not affected when the title field is modified and then emptied,

That sounds abstract, but here's an example: if the title format is `%current_month% | %post_title% | %site_title`, then only the middle part (%post_title%) should change when the title field is modified or emptied; the other two parts should remain unaffected.

**Comment by wpsmort** [comment]
This is tested and is good.

**Pull request merged**